src/request.rs

```diff
@@ -161,16 +189,38 @@ fn get_cache_file_location() -> PathBuf {
     path
 }
 
-async fn request_token(host: &str, ver: ApiVersion, client: &Client) -> Result<String> {
+async fn request_token(
+    host: &str,
+    ver: ApiVersion,
+    creds: &(Option<String>, Option<String>),
+    client: &Client,
+) -> Result<String> {
     let mut auth_url = url_from_host(host, ver.scheme())?;
 
     auth_url
         .path_segments_mut()
         .expect("URL cannot be a base")
         .push("authenticate");
 
-    let username = prompt::simple("User")?;
-    let password = prompt::password("Password")?;
+    // Save token to a file only if credentials weren't supplied from the command line
+    let save_token = creds.0.is_none() && creds.1.is_none();
+
+    let (username, password) = match creds.clone() {
+        (Some(username), Some(password)) => (username, password),
+        (Some(username), None) => {
+            let password = prompt::password("Password")?;
+            (username, password)
+        }
+        (None, Some(password)) => {
```

Comment:
I guess the intended use-case is to give the user another chance to fill in his username if it forgot to pass it?

Reply:
For when only `--password` passed: I don't know, just for completeness sake. It's a consequence of having 2 separate non-mandatory options `--user u --password p` instead of one `--user u:p`.